include/eve/module/real/algorithm/function/regular/generic/count_true.hpp

```diff
@@ -0,0 +1,40 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/concept/conditional.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/detail/top_bits.hpp>
+
+namespace eve::detail
+{
+  EVE_FORCEINLINE std::ptrdiff_t count_true_(EVE_SUPPORTS(cpu_), bool v) noexcept
+  {
+    return v ? 1 : 0;
+  }
+
+  template<value T>
+  EVE_FORCEINLINE std::ptrdiff_t count_true_(EVE_SUPPORTS(cpu_), logical<T> v) noexcept
+  {
+    if constexpr(scalar_value<T>) return v.value() ? 1 : 0;
+    else                          return count_true(eve::detail::top_bits{v});
+  }
```

Comment:
I'm not sure I like this. We don't support ignore for scalar values.
Well - I'm not sure I like scalar values used like this in general.

Reply:
We have logical reduction for scalar logical for genericity purpose.
Now, ignore & logical masking doesn't make sense for scalar, hence why the conditional count_ture is simd_value constrained.